src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesReconciliationService.kt

```diff
@@ -213,13 +212,21 @@ class CaseNotesReconciliationService(
     }.toSortedSet(amendmentComparator),
     caseNoteId,
   )
+
+  // same logic as in nomis prisoner api
+  private fun String.truncateToFixNomisMaxLength(): String = // encodedLength always >= length
+    if (Utf8.encodedLength(this) <= MAX_CASENOTE_LENGTH_BYTES) {
+      this
+    } else {
+      substring(0, MAX_CASENOTE_LENGTH_BYTES - (Utf8.encodedLength(this) - length) - seeDpsReplacement.length) + seeDpsReplacement
+    }
 }
 
 data class MismatchCaseNote(
   val offenderNo: String,
-  val missingFromDps: Set<CommonCaseNoteFields>,
```

Comment:
these two were never actually used in the code, so now hold the id differences between the mappings and NOMIS / DPS